kpi-tracker/src/main.rs

```diff
@@ -793,35 +820,77 @@ async fn process_payday_block(
     if !is_payday_block {
         return Ok(None);
     }
-
-    let tokenomics_info = node
-        .get_tokenomics_info(block_hash)
-        .await
-        .with_context(|| format!("Could not get tokenomics info for block: {block_hash}"))?
-        .response;
-
-    let RewardsOverview::V1 {
-        total_staked_capital,
-        ..
-    } = tokenomics_info else {
-        return Ok(None);
-    };
-
     let mut baker_response = node.get_bakers_reward_period(block_hash).await?.response;
 
     let mut baker_count = 0;
+    let mut open_baker_count = 0;
+    let mut closed_baker_count = 0;
+    let mut open_delegation_recipient_count = 0;
+    let mut closed_for_new_delegation_recipient_count = 0;
     let mut finalizer_count = 0;
+    let mut total_actively_delegated = 0;
+    let mut total_equity_capital = 0;
+    let mut delegating_accounts = HashSet::new();
     while let Some(baker) = baker_response.try_next().await? {
         baker_count += 1;
         if baker.is_finalizer {
             finalizer_count += 1;
         }
+        total_actively_delegated += baker.delegated_capital.micro_ccd() as i64;
+        total_equity_capital += baker.equity_capital.micro_ccd() as i64;
+
+        let pool_info = node
+            .get_pool_info(block_hash, baker.baker.baker_id)
```

Comment:
This can be wrong I think. The baker_id's you have are for the reward period. But now you are querying for the **current** value of the pool. A baker that is currently active might have been deregistered already so I think this call can fail.

Reply:
Being open or closed is a property of the current baker list. You can get this list using `get_baker_list` endpoint.